src/Marten/IRetryPolicy.cs

```diff
@@ -66,4 +67,121 @@ public Task<TResult> ExecuteAsync<TResult>(Func<Task<TResult>> operation, Cancel
             return operation();
         }
     }
+
+    /// <summary>
+    /// Default retry policy, which accounts for <see cref="NpgsqlException.IsTransient"/>.
+    /// </summary>
+    /// <remarks>
+    /// Based on example https://martendb.io/documentation/documents/advanced/retrypolicy/ by Joona-Pekka Kokko.
+    /// </remarks>
+    public sealed class DefaultRetryPolicy: IRetryPolicy
+    {
+        private readonly int _maxTries;
+        private readonly Func<Exception, bool> _filter;
+        private readonly Func<int, TimeSpan> _sleep;
+
+        private static readonly Func<Exception, bool> DefaultFilter = x => x is NpgsqlException npgsqlException && npgsqlException.IsTransient;
+        private static readonly Func<int, TimeSpan> DefaultSleep = x => TimeSpan.FromSeconds(x);
+
+        private DefaultRetryPolicy(int maxTries, Func<Exception, bool> filter, Func<int, TimeSpan> sleep)
```

Comment:
I understand the code has been adapted based on what @jokokko put in the docs. For the value of `maxTries`, user will typically put the exact number of retries i.e. for once, it 1, for twice, it is 2. 

With the current implementation, we are expecting the user to pass the value incremented by 1 which could lead to a bit of confusion. I would suggest that we change the code logic internally as `catch (Exception e) when (tries++ <= _maxTries && _filter(e)) {}` rather than `catch (Exception e) when (++tries < _maxTries && _filter(e)) {}` which requires the value to be incremented by 1 for the condition to work. 
In summary,  the change in code is `++tries` to `tries++`.  Now, user can pass the exact number of retries.

Reply:
Sounds very reasonable, tweaked code with https://github.com/JasperFx/marten/pull/1543/commits/a3f8a7e2bdffb7404795307c33487e65495e0b0f